recipe/meta.yaml

```diff
@@ -13,43 +14,243 @@ source:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - sqlalchemy >=1.3.0,!=1.3.4,<1.4
+        - sqlalchemy-utils
+        - pydot
+        - networkx
+        - flask >=1.0.2
+        - flask-sqlalchemy
+        - pandas
+        - plotly
+        - python-daemon
+    test:
+      imports:
+        - parsl
+        - parsl.monitoring
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-aws
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - boto3
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-kubernetes
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - kubernetes
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+# oauth-ssh not on conda-forge
+#  - name: {{ name }}-with-oauth_ssh
+#    build:
+#      noarch: python
+#    requirements:
+#      host:
+#        - python >=3.6
+#      run:
+#        - python >=3.6
+#        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+#        - oauth-ssh >=0.9
+#    test:
+#      imports:
+#        - parsl
+#      commands:
+#        - pip check
+#      requires:
+#        - pip
+
+  - name: {{ name }}-with-extreme_scale
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - mpi4py
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-google_cloud
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - google-auth
+        - google-api-python-client
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-gssapi
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - python-gssapi
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+# azure 3.0.0 was not built for linux
+#  - name: {{ name }}-with-azure
+#    build:
+#      noarch: python
+#    requirements:
+#      host:
+#        - python >=3.6
+#      run:
+#        - python >=3.6
+#        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+#        - azure <=4
```

Comment:
This is going to be tough to support.  `azure` 3.0.0 only supports py <= 3.6, and conda-forge has dropped support for anything before python 3.7.

The `azure` package is also not being maintained well so I'll try to get myself added and clean that up, but this will only help if we can relax this constraint in a future `parsl` release.

Reply:
Okay, looking into this further, I'm seeing (https://pypi.org/project/azure/):
> Starting with v5.0.0, this package is deprecated. Please install the service specific packages prefixed by azure needed for your application.

So it seems like the upstream repo really needs to redefine this extra.

Reply:
@xylar, `azure` is one of those optional modules that are poorly maintained on our side as well. We could very well just skip it. I can make a note to deprecate it in our docs.